islandora_scholar.drush.inc

```diff
@@ -118,3 +134,147 @@ EOQ;
+ */
+function islandora_scholar_fix_doi_html_in_mods_batch_operation($action, &$context) {
+  $batch_size = 10;
+  /* The islandora_doi_crossref_translator, which introduced the bug that this
+  fixes, is part of the DOI importer, which is hard-coded to
+  import only ir:citationCModel objects. If the repository manager has
+  changed the content models of imported objects, they are responsible for
+  finding and fixing any objects that need it.
+  */
+  $query = <<<EOQ
+SELECT ?pid FROM <#ri>
+WHERE {
+  ?pid <fedora-model:hasModel> <info:fedora/ir:citationCModel> ;
+       <fedora-view:disseminates> ?ds .
+  ?ds <fedora-view:disseminationType> <info:fedora/*/MODS> .
+}
+ORDER BY ?pid
+EOQ;
+  $connection = islandora_get_tuque_connection();
+
+  $sandbox = &$context['sandbox'];
+  if (!isset($sandbox['offset'])) {
+    $sparql_count = $connection->repository->ri->countQuery($query, 'sparql');
+    $sandbox['offset'] = 0;
+    $sandbox['total'] = $sparql_count;
+    if ($sandbox['total'] === 0) {
+      return;
+    }
+  }
+
+  $context['message'] = t('Processing results @start to @end.', array(
+    '@start' => $sandbox['offset'],
+    '@end' => min($sandbox['offset'] + $batch_size, $sandbox['total']),
+  ));
+
+  $offset_start = $sandbox['offset'];
+  $query .= "
+  LIMIT $batch_size
+  OFFSET $offset_start
+  ";
+  $results = $connection->repository->ri->sparqlQuery($query);
+  module_load_include('inc', 'islandora_doi', 'includes/utilities');
+  foreach ($results as $result) {
+    $object = islandora_object_load($result['pid']['value']);
+    $namespace = 'http://www.loc.gov/mods/v3';
+    $doc = new DOMDocument();
+    $doc->loadXML($object['MODS']->content);
+    $needs_work = FALSE;
+    /* The crossref metadata spec dictates that 'Face markup' can only be
+    present in title, subtitle, original_language_title, and
+    unstructured_citation (not implemented in the doi translator).
+    src: https://support.crossref.org/hc/en-us/articles/214532023
+    */
+    foreach (array('title', 'subtitle') as $local_name) {
+      $dubious_nodes = $doc->getElementsByTagNameNS($namespace, $local_name);
+      foreach ($dubious_nodes as $node) {
+        if ($node->childNodes->length > 1) {
+          $needs_work = TRUE;
+        }
+      }
+    }
+    if ($needs_work) {
+      // Dump the original MODS to a backup file.
+      $filename = 'MODS-' . str_replace(':', '_', $object->id) . '-' . format_date(time(), 'custom', 'Y-m-d-His') . '.xml';
+      $mods_file = file_save_data($doc->saveXML(), "temporary://$filename");
+      $mods_file->status &= ~FILE_STATUS_PERMANENT;
+      file_save($mods_file);
+      drush_print("\n=========== Fixing {$object->id}. ========================================\n");
+      drush_print("Old MODS datastream saved to temporary://$filename");
+      module_load_include('inc', 'islandora_doi', 'includes/utilities');
+      foreach (array('title', 'subtitle') as $local_name) {
```

Comment:
Idem, i.e. [that line](https://github.com/Islandora/islandora_scholar/pull/288/files#diff-d342542a4ca9df72fe6804cad9aa55dbR256) should probably read
```php
      foreach (array('title', 'subtitle', 'original_language_title') as $local_name) {
```

Reply:
@d-r-p @rosiel Same comments as the other code review w/r/t `subtitle` vs `subTitle`.

Reply:
@bryjbrown++
Tested and camel case (right way) v/s lower case gives me

DOMNodeList Object
(
    [length] => 1
)
DOMNodeList Object
(
    [length] => 0
)

, so needs to be changed